### PR TITLE
Context menu should have a Edit Description on files

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Closes: enso-org/cloud-v2#1083

Tl;dr: This PR introduces a new menu entry that allows to open edit description dialog from context menu in dashboard. This supposed to work only in cloud.
When you right-click on an item in Cloud Drive, you can choose "Edit description" option to change the description of the selected item.

https://github.com/enso-org/enso/assets/61194245/53e949df-8a31-401c-ba48-52eddad468fa

Context:
See enso-org/cloud-v2#1083 . I decided to open a dialog insted of the sidebar because latter takes to much time and effort to make it properly.

This Change:

Added new variants for button component(submit & cancel), also - loading state.Added a new dialog that opens when you select "Edit description" in context menu.

Test Plan:

1. We shouldn't allow users to change the description for local files
2. Changes in the Dialog(after save) should reflect in sidebar(Description should update in sidebar)
3. Loading state/Errors should be displayed in dialog.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by PabloBuchu** [comment]
QA 🟢

**Review by PabloBuchu — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx [hunk 1]
```diff
@@ -24,31 +28,44 @@ export interface ButtonProps extends reactAriaComponents.ButtonProps {
...
 /**
  * A button allows a user to perform an action, with mouse, touch, and keyboard interactions.
  */
 export function Button(props: ButtonProps) {
-  const { className, children, icon, ...ariaButtonProps } = props
+  const { className, children, variant, icon, loading = false, ...ariaButtonProps } = props
 
-  const classes = clsx(DEFAULT_CLASSES, DISABLED_CLASSES, FOCUS_CLASSES, ICON_CLASSES)
+  const classes = clsx(DEFAULT_CLASSES, DISABLED_CLASSES, FOCUS_CLASSES, SIZE_CLASSES, {
+    [SUBMIT_CLASSES]: variant === 'submit',
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx]: i'd prefer a lookup here: `CLASSES[variant]`

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx [hunk 2]
```diff
@@ -24,31 +28,44 @@ export interface ButtonProps extends reactAriaComponents.ButtonProps {
...
-    return icon != null ? (
-      <>
-        <div className={EXTRA_CLICK_ZONE_CLASSES}>
-          <SvgMask src={icon} />
-        </div>
-      </>
-    ) : (
-      children
-    )
+  const childrenFactory = (): React.ReactNode => {
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:56]: i'd prefer avoiding the function wrapper here. nested ternaries might not be ideal but i think it'll probably be less noisy than `(): type => {` + `}` + a `()` further down. and probably makes it clearer that this isn't a function that is supposed to be called multiple times.

**Review by somebody1234 — APPROVED** [review]
code looks fine, just a few minor nits

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx (hunk 1 above)
Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx]: Agree!

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx (hunk 2 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:56]: (the function wrapper here is fine of course - but the difference was a *lot* more noticeable previously, when there wasn't a nested ternary)
